python_modules/dagster/dagster/_core/execution/stats.py

```diff
@@ -257,9 +257,16 @@ def _open_attempt(step_key: str, event: EventLogEntry) -> None:
         by_step_key[step_key]["partial_attempt_start"] = event.timestamp
 
     def _close_attempt(step_key: str, event: EventLogEntry) -> None:
```

Comment:
do we need to worry about a duplicate close attempt (for whatever reason - maybe a retry comes in at the same time that the step fails?) incorrectly incrementing this twice?

Reply:
Can that even happen?  I thought that you'd either get a failure event or a retry event...

Reply:
I think now that we are retrying on crashes it’s possible for weird stuff to happen